File: csharp/MapleStory.OpenAPI/Src/Dto/CharacterAndroidEquipmentHairDTO.cs
using Newtonsoft.Json;

namespace MapleStory.OpenAPI.Dto
{
    /// <summary>
    /// 캐릭터 안드로이드 헤어 정보
    /// </summary>
    public class CharacterAndroidEquipmentHairDTO
    {
        /// <summary>
        /// 안드로이드 헤어 명
        /// </summary>
        [JsonProperty("hair_name")]
        public string HairName { get; set; }

        /// <summary>
        /// 안드로이드 헤어 베이스 컬러
        /// </summary>
        [JsonProperty("base_color")]
        public string BaseColor { get; set; }

        /// <summary>
        /// 안드로이드 헤어 믹스 컬러
        /// </summary>
        [JsonProperty("mix_color")]
        public string MixColor { get; set; }

        /// <summary>
        /// 안드로이드 헤어 믹스 컬러의 염색 비율
        /// </summary>
        [JsonProperty("mix_rate")]
        public string MixRate { get; set; }
    }
}

File: java/src/main/java/dev/spiralmoon/maplestory/api/msea/dto/character/CharacterAndroidEquipmentHairDTO.java
package dev.spiralmoon.maplestory.api.msea.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

/**
 * Android hair information
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterAndroidEquipmentHairDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterAndroidEquipmentHairDTO {

    /**
     * Android hair name
     */
    @JsonProperty("hair_name")
    private String hairName;

    /**
     * Android base hair color
     */
    @JsonProperty("base_color")
    private String baseColor;

    /**
     * Android mixed hair color
     */
    @JsonProperty("mix_color")
    private String mixColor;

    /**
     * Dye ratio for Android mixed hair color
     */
    @JsonProperty("mix_rate")
    private String mixRate;

    /**
     * Freestyle Coupon application status (0:not applied, 1:applied)
     */
    @JsonProperty("freestyle_flag")
    private String freestyleFlag;

    /**
     * Freestyle Coupon application status
     */
    public boolean isFreestyleFlag() {
        return "1".equals(this.freestyleFlag);
    }
}
